### PR TITLE
fix: Add the option to revert to the AI generated rephrase content

### PR DESCRIPTION
**Pull request by muhsin-k**
Fixes https://linear.app/chatwoot/issue/CW-2056/openai-add-the-option-to-revert-to-the-original-text

The AI-powered rephrasing options may not always capture the intended message and suggest random improvements. Therefore, it is better to revert to the original message and manually edit the newly generated message. If the user is unhappy with the content, they have to rewrite the entire message. This pull request adds the option to revert the content using the cmd+z command if the user is not satisfied with the AI-generated content.

### CODE REVIEW

File: app/javascript/dashboard/components/widgets/AIAssistanceButton.vue [hunk 1]
```diff
@@ -134,11 +135,25 @@ export default {
...
   },
+  destroyed() {
+    document.removeEventListener('keydown', this.onKeyDownHandler);
+  },
+
   methods: {
+    onKeyDownHandler(event) {
+      const { metaKey, ctrlKey, key } = event;
+      const shouldRevertTheContent =
+        (metaKey || ctrlKey) && key === 'z' && !!this.initialMessage;
```
Comment (iamsivin) [inline app/javascript/dashboard/components/widgets/AIAssistanceButton.vue:151]: Can you take a look at KeyboardHelpers file `app/javascript/shared/helpers/KeyboardHelpers.js`? If possible, add the `metaKey || ctrlKey` condition there.

**Review by iamsivin — COMMENTED** [review]
@muhsin-k Please see the inline comments.

File: app/javascript/dashboard/components/widgets/AIAssistanceButton.vue [hunk 2]
```diff
@@ -134,11 +135,25 @@ export default {
...
   },
   mounted() {
+    document.addEventListener('keydown', this.onKeyDownHandler);
     if (!this.appIntegrations.length) {
       this.$store.dispatch('integrations/get');
     }
   },
+  destroyed() {
+    document.removeEventListener('keydown', this.onKeyDownHandler);
+  },
```
[the remark this answers is not in the document]
Comment (muhsin-k) [reply app/javascript/dashboard/components/widgets/AIAssistanceButton.vue]: Makes sense.

**Comment by muhsin-k** [comment]
@iamsivin Fixed all the review comments.

**Review by iamsivin — APPROVED** [review]
The code looks good to me, but haven't tested it. I don't have Open AI integrated into my account.

**Review by iamsivin — APPROVED** [review]
Tested locally. Looks good.
https://www.loom.com/share/8c39e6a2ca094217b18e36ed25a74780?sid=267e1940-539a-4aeb-9447-66322882deaa

File: app/javascript/shared/helpers/KeyboardHelpers.js [hunk 3]
```diff
@@ -106,6 +106,10 @@ export const hasPressedCommandPlusKKey = e => {
   return e.metaKey && e.keyCode === 75;
 };
 
+export const hasPressedCommandZKey = e => {
```
Comment (pranavrajs) [inline app/javascript/shared/helpers/KeyboardHelpers.js:109]: Let us not add methods like this. Please use the buildHotKeys methods written below this.

**Review by pranavrajs — CHANGES_REQUESTED** [review]
The implementation could be simplified to 'Cmd+Z' using buildHotKeys methods.

File: app/javascript/shared/helpers/KeyboardHelpers.js (hunk 3 above)
Comment (muhsin-k) [reply app/javascript/shared/helpers/KeyboardHelpers.js]: @pranavrajs Done.

**Review by pranavrajs — APPROVED** [review]

**Pull request merged**